server/src/main/java/org/opensearch/transport/TransportInterceptorRegistry.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.transport;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Class to register transport interceptors
+ */
+public class TransportInterceptorRegistry {
```

Comment:
Lets see if we can combine core and plugin transport interceptors in a single registry

Reply:
Removed the `TransportInterceptorRegistry` and exposed new method in the networkModule to add coreInterceptors so that core Interceptors will be added first and then pluginInterceptors will follow the coreInterceptors.